Development/nmos/api_downgrade.cpp

```diff
@@ -1,11 +1,973 @@
+            },
+            { make_schema_id(is04_versions::v1_1, nmos::types::receiver, U("subscription")), make_schema(R"({
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                  "subscription": {
+                    "description": "Object containing the 'sender_id' currently subscribed to. Sender_id should be null on initialisation.",
+                    "type": "object",
+                    "required": ["sender_id"],
+                    "properties": {
+                      "sender_id": {
+                        "type": ["string", "null"],
+                        "description": "UUID of the Sender that this Receiver is currently subscribed to",
+                        "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
+                        "default": null
+                      }
+                    }
+                  }
+                }
+              })")
+            },
+            { make_schema_id(is04_versions::v1_2, nmos::types::receiver, U("subscription")), make_schema(R"({
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object",
+                "properties": {
+                  "subscription": {
+                    "description": "Object containing the 'sender_id' currently subscribed to. Sender_id should be null on initialisation, or when connected to a non-NMOS Sender.",
+                    "type": "object",
+                    "required": ["sender_id", "active"],
+                    "properties": {
+                      "sender_id": {
+                        "type": ["string", "null"],
+                        "description": "UUID of the Sender that this Receiver is currently subscribed to",
+                        "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
+                        "default": null
+                      },
+                      "active": {
+                        "type": "boolean",
+                        "description": "Receiver is enabled and configured with a Sender's connection parameters",
+                        "default": false
+                      }
+                    }
+                  }
+                }
+              })")
+            }
+        };
+
+        web::json::value load_json_schema(const web::uri& id)
+        {
+            auto found = property_schemas.find(id);
+
+            if (property_schemas.end() == found)
+            {
+                throw web::json::json_exception((_XPLATSTR("schema not found for ") + id.to_string()).c_str());
+            }
+
+            return found->second;
+        }
+
+        // list of property validators to identify the associated property has been changed since the previous version
+        static const web::json::experimental::json_validator& property_changed_validator()
```

Comment:
Minor nit: `property_changed_validator` sounds like something that is used to perform validation every time a property is changed (e.g. as a result of `nmos::modify_resource`). Not sure what is the best name, though!

Reply:
maybe just named it as `property_validator`, using the word changed because I wanted to specify that the validator is only used for those properties which have schema changed between versions.